core/src/main/java/com/squarespace/template/plugins/platform/CommerceFormatters.java

```diff
@@ -466,18 +468,67 @@ private static String defaultSubscriptionPriceString(JsonNode productNode) {
 
       return sb.toString();
     }
+  }
 
-    private static String getMoneyString(JsonNode moneyNode, Context ctx) {
-      if (CommerceUtils.useCLDRMode(ctx)) {
-        Decimal amount = CommerceUtils.getAmountFromMoneyNode(moneyNode);
-        String currencyCode = CommerceUtils.getCurrencyFromMoneyNode(moneyNode);
-        return PluginUtils.formatMoney(amount, currencyCode, ctx.cldr());
-      } else {
-        Decimal legacyAmount = CommerceUtils.getLegacyPriceFromMoneyNode(moneyNode);
-        StringBuilder buf = new StringBuilder();
-        CommerceUtils.writeLegacyMoneyString(legacyAmount, buf);
-        return buf.toString();
+  protected static class SubscriptionPriceFormatter extends BaseFormatter {
+
+    private Instruction template;
+
+    public SubscriptionPriceFormatter() {
+      super("subscription-price", false);
+    }
+
+    @Override
+    public void initialize(Compiler compiler) throws CodeException {
+      String source = loadResource(CommerceFormatters.class, "subscription-price.html");
+      this.template = compiler.compile(source.trim()).code();
+    }
+
+    @Override
+    public void apply(Context ctx, Arguments args, Variables variables) throws CodeExecuteException {
+      Variable var = variables.first();
+      String subscriptionId = args.isEmpty() ? null : args.first();
```

Comment:
I am allowing a `subscriptionId` argument to be passed so we can render whichever subscription the caller wants.

Reply:
Is relying on the order of arguments the convention/standard in template-compiler or is there a way to parse without having to rely on order? (How easy is it for the caller to pass arguments in the wrong order?)

Reply:
From what I've seen so far, arguments just come in as a list so we have to rely on order.